apps/chat/src/components/AppsEditor/Settings/index.tsx

```diff
@@ -0,0 +1,306 @@
+import {
+  ApiApplicationResponseDefault,
+  ApplicationStatus,
+  ApplicationType,
+} from '@/src/types/applications';
+import { Translation } from '@/src/types/translation';
+
+import {
+  ConversationsActions,
+  ConversationsSelectors,
+} from '@/src/store/conversations/conversations.reducers';
+import { useAppDispatch, useAppSelector } from '@/src/store/hooks';
+import {
+  ModelsActions,
+  ModelsSelectors,
+} from '@/src/store/models/models.reducers';
+import { SettingsSelectors } from '@/src/store/settings/settings.reducers';
+import { UISelectors } from '@/src/store/ui/ui.reducers';
+
+import { Chat } from '../../Chat/Chat';
+import { ApplicationView } from './ApplicationView';
+import { CodeAppView } from './CodeAppView';
+import { CustomApplicationEditorView } from './CustomApplicationEditorView';
+import { QuickAppView } from './QuickAppView';
+import {
+  CodeAppFormData,
+  CustomApplicationFormData,
+  QuickAppFormData,
+  getCodeAppDefaultValues,
+  getCustomApplicationDefaultValues,
+  getQuickAppDefaultValues,
+} from './form';
+
+interface Props {
+  schema: ApiDetailedApplicationTypeSchema | null;
+  applicationData: ApiApplicationResponseDefault;
+  type: string;
+}
+
+export const ApplicationSettings: React.FC<Props> = ({
+  applicationData,
+  schema,
+  type,
+}) => {
+  const pythonVersions = useAppSelector(
+    SettingsSelectors.selectCodeEditorPythonVersions,
+  );
+
+  const isAppDeployed =
+    applicationData?.function?.status === ApplicationStatus.DEPLOYED;
+
+  const modelsMap = useAppSelector(ModelsSelectors.selectModelsMap);
+  const modelFromState = applicationData
+    ? modelsMap[applicationData.reference]
+    : null;
+
+  const previewConversationId = useAppSelector(
+    ConversationsSelectors.selectPreviewConversationId,
+  );
+
+  const isConversationInitialized = useAppSelector(
+    ConversationsSelectors.selectInitialized,
+  );
+
+  const areSelectedConversationLoaded = useAppSelector(
+    ConversationsSelectors.areConversationsUploaded,
+  );
+  const areSelectedConversationsLoaded = useAppSelector(
+    ConversationsSelectors.selectAreSelectedConversationsLoaded,
+  );
+  const theme = useAppSelector(UISelectors.selectThemeState);
+  const { t } = useTranslation(Translation.Chat);
+
+  const [previewMode, setPreviewMode] = useState<'half' | 'full' | 'closed'>(
+    schema?.['dial:applicationTypeViewerUrl'] ? 'closed' : 'half',
+  );
+
+  const getDefaultValues = (type: string) => {
+    const defaultValues: Record<
+      string,
+      CustomApplicationFormData | QuickAppFormData | CodeAppFormData | null
+    > = {
+      [ApplicationType.CUSTOM_APP]: getCustomApplicationDefaultValues({
+        app: applicationData,
+      }),
+      [ApplicationType.CODE_APP]: getCodeAppDefaultValues({
+        app: applicationData,
+        runtime: pythonVersions[0],
+      }),
+      ['Quick App']: getQuickAppDefaultValues({
```

Comment:
Quick app settings tab does not open and show infinite loader instead.
When quick app editor open **type** is equals to 'QuickApps', not 'Quick App'. Same for line 130. 
It's better to map this to ApplicationType enum here also.

Reply:
This is a temporary solution until Quick App has a custom editor.